### DISCUSSION TITLE
Nemo Remote Terminal Integration Idea: wouldn't it be better.

### DISCUSSION THREAD

**Discussion by mohd-minhal** [linuxmint/discussions#665] [Ideas]
Problem
When you connect to a server in Nemo (say, via SSH or SFTP), it only treats it like a storage device, which means you can browse and edit files but can’t actually run commands or manage services. To do anything serious, you still have to open a terminal and SSH into the server manually, which is a hassle.

Suggested Solution: "Open in Remote Terminal"
Imagine if, when you’re browsing a server in Nemo, you could just right-click and select "Open in Remote Terminal." This would pop open a terminal that’s already SSH’d into that server, directly in the directory you’re browsing.

Anyway its a great application, second thing I like about linux mint.

**Comment by 100savage** [comment]
You could create an action for this and place it in your /home/USER/.local/share/nemo/actions directory. You would need to modify a couple of the fields for your system. Like the name, Exec and Comment. 

```
[Desktop Entry]
Name=Server name
Exec=ssh name@10.0.0.2
Comment=SSH to system NAME
Terminal=true
Icon=org.gnome.Terminal
Type=Application
Categories=GNOME;GTK;System;TerminalEmulator;
Hidden=false
NoDisplay=false
```
Restart nemo
```
 nemo -q
```
You can now just right click and select the server name.

**Reply by mohd-minhal** [reply]
Perfect dude, it was so simple.

**Reply by 100savage** [reply]
I'm glad this worked out for you.

**Reply by mohd-minhal** [reply]
Before Closing the discussion, I though why not upload the solution,
I used nemo action,

**open_remote_terminal.nemo_action**
```
[Nemo Action]
Name=Open in Remote Terminal
Comment=Open a terminal SSH'd into the current server and directory (or selected folder).
Icon-Name=org.gnome.Terminal
Selection=Any
UriScheme=sftp
Conditions=uri;
Extensions=dir;
Terminal=true
Exec=path/to/open_in_remote_terminal.sh %U
```
**open_in_remote_terminal.sh**
```
#!/bin/bash

URL="$1"
IP=$(echo "$URL" | sed -n 's|sftp://\([a-zA-Z0-9.-]*\).*|\1|p')
REMOTE_PATH=$(echo "$URL" | sed -n 's|sftp://[a-zA-Z0-9.-]*\(.*\)|\1|p')

echo "Connecting to $IP..."
ssh -t root@"$IP" "cd '$REMOTE_PATH' && exec bash"
fi
```
Anyway, thanks for the help.
